### PR TITLE
Slightly improve error messages on sympy failure

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90655

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -869,6 +877,8 @@ def _maybe_guard_eq(self, expr: "sympy.Eq") -> None:
                 except NotImplementedError:
                     pass
             return
+        except RecursionError:
+            raise RuntimeError(f"RecursionError in sympy.solve({lhs} - {rhs}, {free[0]})")
```
Comment (Skylion007) [inline torch/fx/experimental/symbolic_shapes.py:881]:
```suggestion
        except RecursionError as e:
            raise RuntimeError(f"RecursionError in sympy.solve({lhs} - {rhs}, {free[0]})") from e
```

Comment (ezyang) [reply torch/fx/experimental/symbolic_shapes.py:881]: This is actually intentional, usually the RecursionError is long and annoying so I felt like I don't actually want it haha

Comment (Skylion007) [reply torch/fx/experimental/symbolic_shapes.py:881]: Ah, I thought it might be useful for tracking down the recursion error.

**Review by Skylion007 — APPROVED** [review]

**Review by voznesenskym — APPROVED** [review]

**Pull request closed without merging**